Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs

```diff
@@ -100,5 +104,133 @@ public void ApplyEnabledStateCleansUp()
+                Crashes.Instance.OnChannelGroupReady(_mockChannelGroup.Object, string.Empty);
+                _mockChannel.Verify(channel => channel.EnqueueAsync(It.IsAny<ManagedErrorLog>()), Times.Never());
+                Assert.IsFalse(getErrorLogFilesCalled);
+            }
+        }
+
+        [TestMethod]
+        public void OnChannelGroupReadyDoesNotSendPendingCrashesIfNoneExist()
+        {
+            using (ShimsContext.Create())
+            {
+                // Stub get/read/delete error files
+                ShimErrorLogHelper.GetErrorLogFiles = () =>
+                {
+                    return new List<FileInfo>();
+                };
+                Crashes.SetEnabledAsync(true).Wait();
+                Crashes.Instance.OnChannelGroupReady(_mockChannelGroup.Object, string.Empty);
+                _mockChannel.Verify(channel => channel.EnqueueAsync(It.IsAny<ManagedErrorLog>()), Times.Never());
+            }
+        }
+
+        [TestMethod]
+        public void ProcessPendingErrorsExcludesCorruptedFiles()
+        {
+            var corruptedFileName = "corruptedFile";
+            var fileDeletionCount = 0;
+            var expectedLogId = Guid.NewGuid();
+            var actualSentLogIds = new List<Guid>();
+            var removedLogIds = new List<Guid>();
+            using (ShimsContext.Create())
+            {
+                // Stub get/read/delete error files
+                var file = new FileInfo("file");
+                var corruptedFile = new FileInfo(corruptedFileName);
+                _mockChannel.Setup(channel => channel.EnqueueAsync(It.IsAny<ManagedErrorLog>())).Callback<Log>(log => actualSentLogIds.Add(((ManagedErrorLog)log).Id));
+                ShimFileInfo.AllInstances.Delete = (FileInfo info) => fileDeletionCount++;
+                ShimErrorLogHelper.GetErrorLogFiles = () => new List<FileInfo> { file, corruptedFile };
+                ShimErrorLogHelper.RemoveStoredErrorLogFileGuid = (Guid guid) => removedLogIds.Add(guid);
+                ShimErrorLogHelper.ReadErrorLogFileFileInfo = (FileInfo info) =>
+                {
+                    if (info.Name == corruptedFileName)
+                    {
+                        return null;
+                    }
+                    return new ManagedErrorLog
+                    {
+                        Id = expectedLogId
+                    };
+                };
+
+                Crashes.SetEnabledAsync(true).Wait();
+                Crashes.Instance.OnChannelGroupReady(_mockChannelGroup.Object, string.Empty);
+                Assert.AreEqual(actualSentLogIds.Count, 1);
+                Assert.AreEqual(actualSentLogIds[0], expectedLogId);
+                Assert.AreEqual(removedLogIds.Count, 1);
+                Assert.AreEqual(removedLogIds[0], expectedLogId);
+                // Valid error log file couldn't be deleted in this test because the test mocks RemoveStoredErrorLogFileGuid which deletes the file.
```

Comment:
```suggestion

                // Valid error log file couldn't be deleted in this test because the test mocks RemoveStoredErrorLogFileGuid which deletes the file.
```